packages/inter-protocol/src/auction/auctioneer.js

```diff
@@ -97,20 +107,22 @@ const distributeProportionalShares = (
     currencyLeft = AmountMath.subtract(currencyLeft, currPortion);
     const collPortion = floorMultiplyBy(amount, collShare);
     collateralLeft = AmountMath.subtract(collateralLeft, collPortion);
-    transfers.push([currencySeat, seat, { Currency: currPortion }]);
-    transfers.push([collateralSeat, seat, { Collateral: collPortion }]);
+    transferNonEmpty(currPortion, 'Currency', currencySeat, seat);
+    transferNonEmpty(collPortion, 'Collateral', collateralSeat, seat);
   }
 
-  // TODO(#7117) The leftovers should go to the reserve, and should be visible.
-  transfers.push([currencySeat, reserveSeat, { Currency: currencyLeft }]);
+  transferNonEmpty(currencyLeft, 'Currency', currencySeat, reserveSeat);
+
+  if (!AmountMath.isEmpty(collateralLeft)) {
+    // There will be multiple collaterals, so they can't all use the same keyword
```

Comment:
consider moving this comment adjacent to `[collateralKeyword]`

Reply:
done